app/assets/javascripts/admin/addon/components/admin-user-field-item.js

```diff
@@ -1,146 +1,124 @@
-import Component from "@ember/component";
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
 import { schedule } from "@ember/runloop";
 import { service } from "@ember/service";
 import { isEmpty } from "@ember/utils";
 import { tagName } from "@ember-decorators/component";
 import { Promise } from "rsvp";
 import { popupAjaxError } from "discourse/lib/ajax-error";
-import { i18n, propertyEqual } from "discourse/lib/computed";
-import { bufferedProperty } from "discourse/mixins/buffered-content";
-import discourseComputed from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
 import UserField from "admin/models/user-field";
 
 @tagName("")
-export default class AdminUserFieldItem extends Component.extend(
-  bufferedProperty("userField")
-) {
+export default class AdminUserFieldItem extends Component {
   @service adminCustomUserFields;
   @service dialog;
 
-  isEditing = false;
+  @tracked isEditing = false;
+  @tracked fieldType = this.args.userField.field_type;
```

Comment:
This feels _ick_. I suspect the parent should replace the whole `userField` argument instead when successfully saving.